torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -640,7 +649,12 @@ def analyze_kernel_mutations(
     ops = functions[fn_name]
     for op_list in ops.values():
         for op in op_list:
+            # If we encounter an operation with effects that cannot be reliably analyzed
+            # (e.g. `tt.elementwise_inline_asm`), we assume it does not mutate any input parameters.
```

Comment:
We assume it does not mutate any input parameters only if it has `is_pure=True` right?

Reply:
yes, it's more like if is_pure = True, we assume not mutate of input args